src/memfs.rs

```diff
@@ -1280,11 +1280,11 @@ impl Filesystem for MemoryFilesystem {
         );
 
         let read_helper = |content: &Vec<u8>| {
-            if (offset as usize) < content.len() {
-                let read_data = if ((offset + size as i64) as usize) < content.len() {
-                    &content[(offset as usize)..(offset + size as i64) as usize]
+            if offset.cast::<usize>() < content.len() {
+                let read_data = if (offset + size.cast::<i64>()).cast::<usize>() < content.len() {
```

Comment:
offset.cast() + size < content.len()
no need extra cast()

Reply:
size is u32, content.len() is usize. So need two casts

Reply:
`offset.cast::<usize>() + size.cast::<usize>() < content.len()`

Reply:
Done